src/posix/platform/udp.cpp

```diff
@@ -240,7 +240,10 @@ otError otPlatUdpClose(otUdpSocket *aUdpSocket)
     otError error = OT_ERROR_NONE;
     int     fd;
 
-    VerifyOrExit(aUdpSocket->mHandle != nullptr, error = OT_ERROR_INVALID_ARGS);
+    // Here skips further close ops for non Platform UDP socket.
+    // Platform UDP always has valid mHandle when create.
+    VerifyOrExit(aUdpSocket->mHandle != nullptr);
```

Comment:
Whether it's platform UDP or not, returning success should be the best choice here.